File: SmartShopUI/SmartShopUI/src/app/models/checkout/checkout-data.dto.ts
import { CartItemDto } from '../cart/cart-item.dto';
import { ShippingAddressDto } from '../user/shipping-address.dto';
import { PaymentProviderDto } from '../payment/payment-provider.dto';

export interface CheckoutDataDto {
  cartItems: CartItemDto[];
  address: ShippingAddressDto;
  providers: PaymentProviderDto[];
}

File: SmartShopAPI/SmartShopAPI/Models/Dtos/Order/CheckoutDataDto.cs
using SmartShopAPI.Models.Dtos.CartItem;
using SmartShopAPI.Models.Dtos.Payment;
using SmartShopAPI.Models.Dtos.User;

namespace SmartShopAPI.Models.Dtos.Order
{
    public class CheckoutDataDto
    {
        public List<CartItemDto> CartItems { get; set; } = default!;
        public ShippingAddressDto Address { get; set; } = default!;
        public List<PaymentProviderDto> Providers { get; set; } = default!;
    }
}
